### PR TITLE
EventListener to aggregate activities

### PR DESCRIPTION
**Pull request by samsp-msft**
Posting for @Tratcher 

Should not be pulled as-is. Creates an EventListener that will aggregate the events, creating a parent/child based on ActivityID in the events.
Based on conversations, this probably needs to be inverted, and push EventSource Events to ILogger and ensure that they scope correctly.

### CODE REVIEW

File: src/ReverseProxy.Core/Service/Proxy/RoundRobinLoadBalancer.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.ReverseProxy.Core.RuntimeModel;
+
+namespace ReverseProxy.Core.Service.Proxy
+{
+    internal class RoundRobinLoadBalancer
```
Comment (Tratcher) [inline src/ReverseProxy.Core/Service/Proxy/RoundRobinLoadBalancer.cs:11]: Did you mean to include this? It seems unrelated.

File: .github/ISSUE_TEMPLATE/usage-questions.md [hunk 2]
```diff
@@ -0,0 +1,351 @@
+---
+name: Usage Questionaire
```
Comment (Tratcher) [inline .github/ISSUE_TEMPLATE/usage-questions.md:2]: Also unrelated?

Comment (samsp-msft) [reply .github/ISSUE_TEMPLATE/usage-questions.md:2]: yes

File: src/ReverseProxy.Core/Service/Proxy/RoundRobinLoadBalancer.cs (hunk 1 above)
Comment (samsp-msft) [reply src/ReverseProxy.Core/Service/Proxy/RoundRobinLoadBalancer.cs:11]: I shouldn't have done an add --all

**Comment by Tratcher** [comment]
@samsp-msft can this be closed in favor of @MihaZupan's work?
